client/task_runner.go

```diff
@@ -1025,6 +1026,11 @@ func (r *TaskRunner) run() {
 				r.runningLock.Lock()
 				running := r.running
 				r.runningLock.Unlock()
+
+				// Remove from consul before killing the task so that traffic
+				// can be rerouted
+				r.consulCleanup()
```

Comment:
Can you move this to below the `!running` check and then just call `r.consul.RemoveTask(r.alloc.ID, r.task)`